### DISCUSSION TITLE
Why isn't the tenant key added with ->withPivot() on the roles() relationship?

### DISCUSSION THREAD

**Discussion by juliangums** [spatie/laravel-permission#2680] [General]
I noticed I could not find the team id pivot in ->getPivotColumns() or ->pivot and I was wondering why it has not been added like this https://github.com/spatie/laravel-permission/pull/2679/files. Obviously I can see that all checks are failing on there as that seems to mess with the test db but I for my part have just overridden the roles() method in my model the same way and it's all working as expected and I have access to the pivot value now. I am just wondering if this was intentionally done that way and why.

**Comment by erikn69** [comment]
https://github.com/spatie/laravel-permission/pull/2679#issuecomment-2159279444
>this was intentionally done that way and why

This was done in Laravel 6, since that version many features have been added

**Comment by juliangums** [comment]
@erikn69 updated and reopened the pull request. Thanks for looking into this. Will close this discussed then.
